### ISSUE TITLE
Kernel 6.11.0 Compatibility Issue

### ISSUE THREAD

**Issue by Jord-Sudo** [morrownr/rtl8852bu#40]
Hello,
I’m new to using Ubuntu and Linux in general. I’ve been trying to get my Newfast NF-U353 Wi-Fi adapter to work, but currently, I can only connect to the internet through a wired connection. There’s no Wi-Fi option available for me to choose.

I was advised to find a Linux driver for my device, and after asking for help on Ask Ubuntu, I was directed to your repository ([link](https://github.com/morrownr/rtl8852bu/tree/1.19.3)). Following the instructions, I downloaded and compiled the 8852bu driver with some assistance from ChatGPT, but it doesn’t seem to load properly, and I still don’t have the option to connect via Wi-Fi.

Any advice or guidance would be greatly appreciated.

Thank you for your help!

**Comment by morrownr** [comment]
Hi @Jord-Sudo 

Welcome to Linux. I've been using Linux since 1994. Ubuntu is a good distro.

First, I would like to check that this driver is the correct driver. Please run the following command and paste the results:

$ lsusb

Does your system have Secure Boot active?

Here is some more info for you but it does not have to do with the task at hand, but I think it could help.

Here is the Main Menu for my site:

https://github.com/morrownr/USB-WiFi

Recommended reading includes menu items 1 and 2. That should give you a good idea of the state of using USB WiFi Adapters on Linux. There are many good options and things are improving but adapters based on the Realtek WiFi 6 USB chips are something I warn Linux users to stay away from. Realtek simply does a poor job with their USB WiFi support.

**Comment by Jord-Sudo** [comment]
Hello, and thank you for your reply!

I’m glad to hear I’m in the right place for help.

To answer your question, I do not have Secure Boot active—my system is in
legacy mode. That said, I’ve been encountering another issue: since I
installed Ubuntu using a flash drive, the option to boot into Windows is
missing when I press F12 during startup. It seems I’m stuck in Ubuntu for
now and can’t switch back to Windows when needed.

Thank you for the additional reading materials. I’ll definitely take a look.

Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
Bus 001 Device 002: ID 8087:0024 Intel Corp. Integrated Rate Matching Hub
Bus 002 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
Bus 002 Device 002: ID 8087:0024 Intel Corp. Integrated Rate Matching Hub
Bus 002 Device 003: ID c0f4:07b0 USB usb keyboard
Bus 002 Device 004: ID 1c4f:0034 SiGma Micro XM102K Optical Wheel Mouse
Bus 003 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
Bus 003 Device 003: ID 0bda:c832 Realtek Semiconductor Corp. 802.11ax WLAN
Adapter
Bus 004 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hub

**Comment by morrownr** [comment]
> Bus 003 Device 003: ID 0bda:c832 Realtek Semiconductor Corp. 802.11ax WLAN

That tells me that your device is based on the rtl8852cu or rtl8832cu chip so the driver in this repo is not for your adapter. I have been working on a driver for your adapter but it is still private for now... just me and about a dozen testers. You may be able to run it fine. To remove this driver, open a terminal (command line prompt) in your folder with this driver and run the following:

$ sudo sh remove-driver.sh

After reboot, you can delete the folder with this driver.

Now, do you have a github account? And do you want access to the rtl8852/32cu driver?

> To answer your question, I do not have Secure Boot active—my system is in
legacy mode.

Since you are running Ubuntu, secure boot should not have been a problem anyway but, things happen.

> That said, I’ve been encountering another issue: since I installed Ubuntu using a flash drive, the option to boot into Windows is missing when I press F12 during startup.

Can we look at this after we fix the driver problem?

**Comment by Jord-Sudo** [comment]
Using the ls command in my desktop directory, I see two folders: rtl8812au
and rtl8852bu. I believe these are the drivers I recently attempted to
install. I ran the sudo sh remove-driver.sh command while inside both of
these folders, but after rebooting, the folders are still present.

To clarify, if the lsmod | grep 8812au command produces no output, does
that confirm the driver has been successfully uninstalled?

Regarding the rtl8852/32cu driver, yes, I have a GitHub account and would
like access to it—thank you for offering!
Yes, that would be perfect if you could help with my other issue after
fixing the driver problem first.

**Comment by morrownr** [comment]
> Using the ls command in my desktop directory, I see two folders: rtl8812au
and rtl8852bu. I believe these are the drivers I recently attempted to
install.

That is the case because when you clone something with git, it creates the folders.

>  I ran the sudo sh remove-driver.sh command while inside both of
these folders, but after rebooting, the folders are still present.

This is normal. remove-driver.sh uninstalls the driver but does not delete the driver folder for a few reasons. If you no longer need the folders, you can delete them.

> To clarify, if the lsmod | grep 8812au command produces no output, does
that confirm the driver has been successfully uninstalled?

If you get nothing while the rtl8812au based adapter is the usb port, then the driver is likely installed. The install and remove scripts are really pretty solid and if they say the driver was installed or removed successfully, then that is likely true as they have error checking and will tell you if there is a problem in most cases.

> Regarding the rtl8852/32cu driver, yes, I have a GitHub account and would
like access to it—thank you for offering!

I have sent you an invitation in email. Follow the directions and I will see you in the repo.

**Comment by Jord-Sudo** [comment]
The output I get from the from sudo sh remove-driver.sh command in the rtl8812au folder is sh: 0: cannot open remove-driver.sh: No such file.

I have accepted your invitation via email.

**Comment by morrownr** [comment]
> The output I get from the from sudo sh remove-driver.sh command in the rtl8812au folder is sh: 0: cannot open remove-driver.sh: No such file.

That is normal if you are not using one of my repos. If that is the case, you'll have to following the uninstallation instructions that came with that driver.

> I have accepted your invitation via email.

I'll see you there.

FYI: So that you don't think this is how Linux is all the time, you have an adapter that has a Realtek chip in it. If you have an adapter with a Mediatek chip, then all you would have needed to do is plug it in. We can discuss this. If you can still return your adapter, if you do return it, I can point you to an adapter that will be very easy to use on Linux.

**Comment by Jord-Sudo** [comment]
I am not able to return the Wi-Fi adapter that I am trying to install at
the moment.

If I’m still having no luck with the repo you invited me to then I will
have no choice to purchase a new one.

How do I install the driver with the invitation you gave me?

**Comment by morrownr** [comment]
Go to the repo. Here is the url:

https://github.com/morrownr/rtl8852bu-20240418

Scroll down. Read the README.md.

Let me know if you have problems.

**Comment by morrownr** [comment]
Rats! I send you the wrong url... standby.

Here is the correct one:

https://github.com/morrownr/rtl8852cu-20240510

**Comment by Jord-Sudo** [comment]
From this command.
git clone https://github.com/morrownr/rtl8852cu-20240510.git

I got this output.
Cloning into 'rtl8852cu-20240510'...
Username for 'https://github.com': Jord-Sudo
Password for ***@***.***':
remote: Support for password authentication was removed on August 13, 2021.
remote: Please see
https://docs.github.com/get-started/getting-started-with-git/about-remote-repositories#cloning-with-https-urls
for information on currently recommended modes of authentication.
fatal: Authentication failed for '
https://github.com/morrownr/rtl8852cu-20240510.git/'

Do i now need to install this for it to work?
https://github.com/git-ecosystem/git-credential-manager/blob/main/README.md

**Comment by morrownr** [comment]
> Support for password authentication was removed on August 13, 2021.

Yes, for many things, a password is not enough. Github wants the Token. It is called a Github Token.

See if this helps:

https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens

**Comment by Jord-Sudo** [comment]
How do I now give the token I just created access to your repo?
There doesn't seem to be an option for me to be able to.

I get this output now.
remote: Write access to repository not granted.
fatal: unable to access 'https://github.com/morrownr/rtl8852cu-20240510.git/':
The requested URL returned error: 403

Ha my bad if I don't seem to be getting there straight away. We are getting
there in just small steps.

**Comment by morrownr** [comment]
Remember when you put the url for my repo in and it ask for id and password? Put the token in the place where it asks for the password. You won't use the password.

This is silliness.

**Comment by Jord-Sudo** [comment]
Yeah I did. I copied and pasted the token when asked for password, but
still get that output.

**Comment by Jord-Sudo** [comment]
I'm still hitting a brick wall here. Do you need to give me read or write
access because it's a private repo?

This is the output i'm getting.
remote: Write access to repository not granted.
fatal: unable to access 'https://github.com/morrownr/rtl8852cu-20240510.git/':
The requested URL returned error: 403

**Comment by morrownr** [comment]
@Jord-Sudo 

Well, I have run into that same brick wall... the one where it is not obvious what to do next.

There are already about a dozen people in the repo testing and I have not run into this.

I have retried the link:

https://github.com/morrownr/rtl8852cu-20240510/

It works for me so the url must be good. Now if you could post the exact steps you are taking once you tell your browser to go to the site above.

I did check in the repo and you have the same rights as the others that are already there.

**Comment by Jord-Sudo** [comment]
Ok, so I made another token to see if that was the problem.
Gave it Repository access to All repositories.
Repository permissions. I changed all permissions to read and write, the
ones I couldn't change to both, I changed the permission to read only = 27
permissions for all of your repositories.
Account permissions. I did the same with account permissions = 16 Account
permissions.

Currently stuck on step 6 in the README.md
I copy and paste the git clone
https://github.com/morrownr/rtl8852cu-20240510.git into my terminal and get
this output.
Cloning into 'rtl8852cu-20240510'...
Username for 'https://github.com': Jord-Sudo
Password for ***@***.***':
remote: Write access to repository not granted.
fatal: unable to access 'https://github.com/morrownr/rtl8852cu-20240510.git/':
The requested URL returned error: 403

"Now if you could post the exact steps you are taking once you tell your
browser to go to the site above."
I hope that was what you were asking for.

**Comment by morrownr** [comment]
@Jord-Sudo 

Never seen this before. When I first got my token I do not remember having to set any of those settings. This is crazy.

I cannot explain this. Maybe I should delete you from the repo and then add you again. Let me try.

**Comment by Jord-Sudo** [comment]
Damn ok, let me know what you want me to do next.

**Comment by morrownr** [comment]
@Jord-Sudo 

I deleted you and then invited you again. Let's see what happens.

**Comment by morrownr** [comment]
> Am I going to have to get another wifi adapter?

That is up to you. To aid you toward that end, below is the Main Menu for this site. Recommend reading menu items 1 and 2. Menu item 1 is background and recommendations regarding USB WiFi Adapters and Linux. Menu item 2 is a detailed list with many adapters that are known to work well with Linux... The Plug and Play List.

https://github.com/morrownr/USB-WiFi

So that we can say that this problem did not defeat us, we do need to solve the problem. For now, I am out of answers.

**Comment by Jord-Sudo** [comment]
The plug in a play list sounds good to me.
A wired connection is not realistic with my computer set up, it was only
temporary until I sorted the wifi problem out.

My other problem was not being able to boot back into windows when I
wanted. Would you be able to help with this?

**Comment by Jord-Sudo** [comment]
"That list has adapters that work when you plug them in. No fighting with
drivers. I can give you recommendations from the list if you give me an
idea of what you need."

I would just need a good decent wifi connection to browse and occasionally
stream.
I just looked at the list and the only one I found available was
https://www.amazon.co.uk/EDUP-Wireless-802-11AX-Tri-Band-Compatible-Black/dp/B0DFYBZSR6
(I'm based in the UK) Can you confirm that this is one of the updaters on
the plug in a play list?

"Let me see... you are running Ubuntu 24.10?"

Yes, that's correct.

"Do you remember which items that you selected during installation? I may
need to boot up an installation flash driver to ask you specifics. When you
got to the point where you select the type of installation, I hope you did
not say "use the entire disk" as that would have thrown Windows into the
trash can. That would not be good if you still want to run Windows. In that
case, we would have to look at reinstalling Windows and then reinstalling
Ubuntu (carefully). It also could be a case your startup menu ... you do
see a startup menu when you boot the system?"

I remember choosing the option to run windows alongside Ubuntu.
I also looked up a command to check if windows was detected on the computer
and it was there.

**Comment by morrownr** [comment]
> I just looked at the list and the only one I found available was...

Think that Edup is spot on. While they did not include the Edup model number in the ad, it did show the following down a few pages:

Chipset	MTK7921AU

You should be happy with that adapter and it should just work when you plug it into Ubuntu.

> I remember choosing the option to run windows alongside Ubuntu.

Okay, that sounds good. Now to figure out why you are not seeing a menu on startup... one of the items in the menu should be something like `Boot to Windows` ?

Do you have a setup where F12 brings up a system boot menu?

How it should be is everytime you cold boot the system, a menu should appear for a few seconds that will allow you to select which is to boot. It is not clear why this is not happening.

**Comment by Jord-Sudo** [comment]
“You should be happy with that adapter and it should just work when you
plug it into Ubuntu”

I’ll give that adapter a go then and hopefully all my Wi-Fi problems should
go away ha.

“Do you have a setup where F12 brings up a system boot menu?”

There is currently no windows option to choose from.

When pressing F12 on startup I get this menu.


Boot mode is set to: LEGACY; Secure Boot: OFF


LEGACY BOOT:

ST500DM002 - 1BD142 (this option is selected)

CD/DVD/CD-RW Drive

Onboard NIC


OTHER SETUP

BIOS Setup

Diagnostics

Change Boot Mode Settings


When selecting BIOS Setup

Then Boot Sequence to see what are my options to boot into I only have the
following options.


Boot Sequence

Diskette Drive

ST500DM002 - 1BD142

USB Storage Device

CD/DVD/CD-RW Drive

Onboard NIC


Boot List Option

Legacy (this option is selected)

UEFI

**Comment by morrownr** [comment]
G'day,

> I’ll give that adapter a go then and hopefully all my Wi-Fi problems should
go away ha.

Let me know what happens. It should be a better experience. 

> There is currently no windows option to choose from.

I see. It appears that you have one drive that is a Seagate and is 500 gb. The system is in legacy mode.

So, the only menu you are seeing is the one given with F12.

The installation of Ubuntu alongside an existing installation of Windows should have generated a Grub Menu, which would appear on reboot, showing Ubuntu at the top, then advanced options and followed by a line mentioning Windows Boot Manager. Something has not worked correctly. This is odd because Ubuntu is a very stable distro.

To manually start the Grub menu during boot, immediately hold down on the right Shift key as soon as you turn the system on. That should trigger the Grub menu. See if that works and what the menu items are.

**Comment by Jord-Sudo** [comment]
“G'day”

Hey!

“To manually start the Grub menu during boot, immediately hold down on the
right Shift key as soon as you turn the system on. That should trigger the
Grub menu. See if that works and what the menu items are.”

Haha well that worked thanks. (We wasn’t entirely defeated)

It was the left shift key that worked by the way.
Not the right. (Just for future reference)

I looked up how to get the grub menu so appear automatically, so I don’t
have to press the shift key every time.

Does this sound right to your knowledge?

*Persistent GRUB Display*

If the GRUB menu doesn't show by default and you want it to appear at every
boot:

   - Edit the */etc/default/grub* file by running:

   bash
   Copy code
   sudo nano /etc/default/grub

   - Look for the line GRUB_TIMEOUT=0 or GRUB_HIDDEN_TIMEOUT=0 and change
   the value to a non-zero number (e.g., GRUB_TIMEOUT=5).
   - Save the file and update GRUB:

   bash
   Copy code
   sudo update-grub

   - The menu will now appear for the specified number of seconds at every
   boot.

By following these steps, you can manually or persistently access the GRUB
menu during system startup.rsistent GRUB Display


If the GRUB menu doesn’t show by default and you want it to appear at every
boot:

• Edit the /etc/default/grub file by runni

**Comment by morrownr** [comment]
@Jord-Sudo 

> It was the left shift key that worked by the way. Not the right. (Just for future reference)

It has been a long time since I needed to do that but keep in mind, it can be a different key depending on whether you are using BIOS or UEFI. Glad you figured that out. You should not have needed to do that so something was not quite right with the Ubuntu installer. 

So, we won a battle. Another tester said he had problems getting into the repo so I asked him what he did and I am waiting on an answer.

Since you are new to Linux, here is some advice: Yes, the internet is full of all kinds of help for Linux but when searching, I recommend you start the search with settings to see only the last year. It can be expanded if necessary. The reason is that Linux is continuously being improved and a lot of old information is not longer valid.

**Comment by Jord-Sudo** [comment]
Hey

"So, we won a battle. Another tester said he had problems getting into the
repo so I asked him what he did and I am waiting on an answer."

Any luck?

**Comment by morrownr** [comment]
> Any luck?

Yes, the guy said he needed to establish the new 2 factor login. That is relatively new. See if you can set that up and we can try again.

**Comment by Jord-Sudo** [comment]
Did all that and is still not working.

Cloning into 'rtl8852cu-20240510'...
Username for 'https://github.com': Jord-Sudo
Password for ***@***.***':
remote: Write access to repository not granted.
fatal: unable to access 'https://github.com/morrownr/rtl8852cu-20240510.git/':
The requested URL returned error: 403

Thanks for the help anyway.

**Comment by morrownr** [comment]
Not sure what is going on.

I can zip the repo and give you access via Google Docs. Do you have a Google account?

**Comment by Jord-Sudo** [comment]
I don't have a Google account.

We can try that if you want.

**Comment by morrownr** [comment]
I just compressed the repo and it is 19 MB. Unless things have expanded, this file size may be too big to attach to an email. I will try it if you want to post an email address.

**Comment by morrownr** [comment]
@Jord-Sudo 

I opened the repo to the public today:

https://github.com/morrownr/8821cu-20210916
